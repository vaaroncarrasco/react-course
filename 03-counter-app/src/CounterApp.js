import React, { useState } from 'react'
// rafcp - > shortcut

// ? Functional Components
// * -> Functional Component has a props{} you can destructure
// props -> destructued -> props.value -> { value }
const CounterApp = ({ value = 10 }) => {

    // * useState() returns state[] array -> [value, setValue]
    const [ counter, setCounter ] = useState( value );

    const handleAdd = e => setCounter(counter + 1)
    const handleSubtract = e => setCounter(counter - 1)
    const handleReset = e => setCounter(value)

    // * In => when first argument is sent as first argument to a callback -> simplify function without ()
    // <button onClick={ (e) => handleAdd(e) } >+1</button>
    // * Without () because its a callback, it is called when needed, not right away
    // <button onClick={ handleAdd } >+1</button>

    return (
        <>
            {/* printing objects */}
            {/* <pre> { JSON.stringify(object, null, 3) } </pre> */}
            <h1>CounterApp</h1>
            <h2>{ counter }</h2>
            <button onClick={ handleAdd }>+1</button>
            <button onClick={ handleReset }>Reset</button>
            <button onClick={ handleSubtract }>-1</button>
        </>
    )

    // return (
    //     <Fragment>
    //         <h1>Hola mundo</h1>
    //         <p>Mi primera app</p>
    //     </Fragment>
    // );
}

export default CounterApp


