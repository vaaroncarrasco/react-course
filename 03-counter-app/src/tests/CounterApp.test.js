import React from "react";
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import CounterApp from "../CounterApp";

describe('testing <CounterApp />', () => {

    let wrapper = shallow( <CounterApp />);

    // * To reset component before each test
    beforeEach( () => {
        wrapper = shallow( <CounterApp />);
    });

    test('should show <CounterApp /> & match snapshot', () => {
        expect( wrapper ).toMatchSnapshot();
    });


    test('should show value 100', () => {
        const wrapper = shallow( <CounterApp value={ 100 }/>);
        const counter = wrapper.find('h2').text();
        expect( wrapper ).toMatchSnapshot();
        expect( counter ).toBe('100');
    });

    test('should increment w/ btn +1', () => {
        wrapper.find('button').at(0).simulate('click');
        const counterText = wrapper.find('h2').text();
        expect( counterText ).toBe('11');
    });

    test('should decrement w/ btn -1', () => {
        wrapper.find('button').at(2).simulate('click');
        const counterText = wrapper.find('h2').text();
        expect( counterText ).toBe('9');
    });

    test('should reset value w/ btn', () => {
        const wrapper = shallow( <CounterApp value={ 69 }/>);

        wrapper.find('button').at(0).simulate('click');
        wrapper.find('button').at(0).simulate('click');
        wrapper.find('button').at(1).simulate('click');

        const counterText = wrapper.find('h2').text();

        expect( counterText ).toBe('69');
    });

});
