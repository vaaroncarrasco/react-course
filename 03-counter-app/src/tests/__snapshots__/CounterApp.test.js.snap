// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing <CounterApp /> should show <CounterApp /> & match snapshot 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h2>
    10
  </h2>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
</Fragment>
`;

exports[`testing <CounterApp /> should show value 100 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h2>
    100
  </h2>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
</Fragment>
`;
